File: quizz/travez/u3/create_speakers/utils/MongoDB_controller.py
import os
import sys
from pymongo import MongoClient

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from espe_edu_ec_speakers_controller.speaker_controller import speaker_controller
from espe_edu_ec_speakers_model.speaker import speaker

class MongoDB_controller():
    def create_speaker(self, speaker):
        client = MongoClient("mongodb+srv://ismael:ismael@cluster0.tayl6re.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0")
        database = client["oop"]
        collection = database["speakers"]
        
        document = {
            "id": speaker.id,
            "name": speaker.name,
            "price": speaker.price,
            "iva": speaker_controller.calculate_iva(speaker.price)
        }
        collection.insert_one(document)
        client.close()

File: exams/Vaca/U3/ExamUnit3/controller/MongoDBController.js
const { MongoClient } = require('mongodb');

class MongoDBController {
    static async saveSortedArray(sortType, sortedArray) {
        const uri = "mongodb+srv://vaca:vaca@cluster0.42hdahs.mongodb.net/"; 
        const client = new MongoClient(uri);

        try {
            await client.connect();
            const database = client.db('OOPEXAM3');
            const collection = database.collection('Array');
            const doc = {
                sortType: sortType,
                sortedArray: sortedArray,
                timestamp: new Date()
            };
            await collection.insertOne(doc);
            console.log("Document inserted successfully");
        } finally {
            await client.close();
        }
    }
}

module.exports = MongoDBController;
